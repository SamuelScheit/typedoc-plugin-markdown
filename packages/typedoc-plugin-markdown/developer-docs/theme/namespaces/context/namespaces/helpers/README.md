[typedoc-plugin-markdown v4.0.3](../../../../../README.md) / [theme](../../../../README.md) / [context](../../README.md) / helpers

# Namespace: helpers

The `helpers` namespace holds the helpers for the theme and are smaller utility functions that return snippets or text or other data transformations.

Please note that partials:

* Take a `model` param (that references a specific TypeDoc model) and an `options` param if required.
* Can reference other helpers but should not reference partials.
* Can return strings or other models.

## Table of Contents

* [getCommentParts()](#getcommentparts)
* [isFile()](#isfile)
* [getDeclarationType()](#getdeclarationtype)
* [getDescriptionForReflection()](#getdescriptionforreflection)
* [getFlattenedDeclarations()](#getflatteneddeclarations)
* [getGroupIndexList()](#getgroupindexlist)
* [getGroupIndexTable()](#getgroupindextable)
* [getGroupIndex()](#getgroupindex)
* [getHierarchyType()](#gethierarchytype)
* [getKeyword()](#getkeyword)
* [getModifier()](#getmodifier)
* [getParameterDefaultValue()](#getparameterdefaultvalue)
* [getProjectName()](#getprojectname)
* [getPropertyDefaultValue()](#getpropertydefaultvalue)
* [getReflectionFlags()](#getreflectionflags)
* [getReturnType()](#getreturntype)
* [isGroupKind()](#isgroupkind)
* [useTableFormat()](#usetableformat)

## getCommentParts()

```ts
function getCommentParts(this, model): string
```

### Parameters

| Parameter | Type                                                                                    |
| --------- | --------------------------------------------------------------------------------------- |
| `this`    | [`MarkdownThemeContext`](../../../../classes/MarkdownThemeContext.md)                   |
| `model`   | [`CommentDisplayPart`](https://typedoc.org/api/types/Models.CommentDisplayPart.html)\[] |

### Returns

`string`

### Defined in

[packages/typedoc-plugin-markdown/src/theme/context/helpers/get-comment-parts.ts:6](https://github.com/typedoc2md/typedoc-plugin-markdown/blob/a350891d3362a78bb12907d480645f9c5cefd0d6/packages/typedoc-plugin-markdown/src/theme/context/helpers/get-comment-parts.ts#L6)

***

## isFile()

```ts
function isFile(file): boolean
```

### Parameters

| Parameter | Type     |
| --------- | -------- |
| `file`    | `string` |

### Returns

`boolean`

### Defined in

[packages/typedoc-plugin-markdown/src/theme/context/helpers/get-comment-parts.ts:64](https://github.com/typedoc2md/typedoc-plugin-markdown/blob/a350891d3362a78bb12907d480645f9c5cefd0d6/packages/typedoc-plugin-markdown/src/theme/context/helpers/get-comment-parts.ts#L64)

***

## getDeclarationType()

```ts
function getDeclarationType(model): undefined | SomeType
```

### Parameters

| Parameter | Type                                                                                         |
| --------- | -------------------------------------------------------------------------------------------- |
| `model`   | [`DeclarationReflection`](https://typedoc.org/api/classes/Models.DeclarationReflection.html) |

### Returns

`undefined` | [`SomeType`](https://typedoc.org/api/classes/Models.SomeType.html)

### Defined in

[packages/typedoc-plugin-markdown/src/theme/context/helpers/get-declaration-type.ts:3](https://github.com/typedoc2md/typedoc-plugin-markdown/blob/a350891d3362a78bb12907d480645f9c5cefd0d6/packages/typedoc-plugin-markdown/src/theme/context/helpers/get-declaration-type.ts#L3)

***

## getDescriptionForReflection()

```ts
function getDescriptionForReflection(this, model): null | string
```

### Parameters

| Parameter | Type                                                                                         |
| --------- | -------------------------------------------------------------------------------------------- |
| `this`    | [`MarkdownThemeContext`](../../../../classes/MarkdownThemeContext.md)                        |
| `model`   | [`DeclarationReflection`](https://typedoc.org/api/classes/Models.DeclarationReflection.html) |

### Returns

`null` | `string`

### Defined in

[packages/typedoc-plugin-markdown/src/theme/context/helpers/get-description-for-reflection.ts:4](https://github.com/typedoc2md/typedoc-plugin-markdown/blob/a350891d3362a78bb12907d480645f9c5cefd0d6/packages/typedoc-plugin-markdown/src/theme/context/helpers/get-description-for-reflection.ts#L4)

***

## getFlattenedDeclarations()

```ts
function getFlattenedDeclarations(model, options?): DeclarationReflection[]
```

### Parameters

| Parameter                    | Type                                                                                            |
| ---------------------------- | ----------------------------------------------------------------------------------------------- |
| `model`                      | [`DeclarationReflection`](https://typedoc.org/api/classes/Models.DeclarationReflection.html)\[] |
| `options`?                   | `object`                                                                                        |
| `options.includeSignatures`? | `boolean`                                                                                       |

### Returns

[`DeclarationReflection`](https://typedoc.org/api/classes/Models.DeclarationReflection.html)\[]

### Defined in

[packages/typedoc-plugin-markdown/src/theme/context/helpers/get-flattened-declarations.ts:3](https://github.com/typedoc2md/typedoc-plugin-markdown/blob/a350891d3362a78bb12907d480645f9c5cefd0d6/packages/typedoc-plugin-markdown/src/theme/context/helpers/get-flattened-declarations.ts#L3)

***

## getGroupIndexList()

```ts
function getGroupIndexList(this, children): string
```

### Parameters

| Parameter  | Type                                                                                                                       |
| ---------- | -------------------------------------------------------------------------------------------------------------------------- |
| `this`     | [`MarkdownThemeContext`](../../../../classes/MarkdownThemeContext.md)                                                      |
| `children` | [`DeclarationReflection`](https://typedoc.org/api/classes/Models.DeclarationReflection.html)\[] \| `DocumentReflection`\[] |

### Returns

`string`

### Defined in

[packages/typedoc-plugin-markdown/src/theme/context/helpers/get-group-index-list.ts:6](https://github.com/typedoc2md/typedoc-plugin-markdown/blob/a350891d3362a78bb12907d480645f9c5cefd0d6/packages/typedoc-plugin-markdown/src/theme/context/helpers/get-group-index-list.ts#L6)

***

## getGroupIndexTable()

```ts
function getGroupIndexTable(
   this, 
   children, 
   kind?): string
```

### Parameters

| Parameter  | Type                                                                                                                       |
| ---------- | -------------------------------------------------------------------------------------------------------------------------- |
| `this`     | [`MarkdownThemeContext`](../../../../classes/MarkdownThemeContext.md)                                                      |
| `children` | [`DeclarationReflection`](https://typedoc.org/api/classes/Models.DeclarationReflection.html)\[] \| `DocumentReflection`\[] |
| `kind`?    | [`ReflectionKind`](https://typedoc.org/api/enums/Models.ReflectionKind-1.html)                                             |

### Returns

`string`

### Defined in

[packages/typedoc-plugin-markdown/src/theme/context/helpers/get-group-index-table.ts:10](https://github.com/typedoc2md/typedoc-plugin-markdown/blob/a350891d3362a78bb12907d480645f9c5cefd0d6/packages/typedoc-plugin-markdown/src/theme/context/helpers/get-group-index-table.ts#L10)

***

## getGroupIndex()

```ts
function getGroupIndex(group): any
```

### Parameters

| Parameter | Type                                                                                                                                                                     |
| --------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
| `group`   | [`ReflectionCategory`](https://typedoc.org/api/types/Models.ReflectionCategory.html) \| [`ReflectionGroup`](https://typedoc.org/api/classes/Models.ReflectionGroup.html) |

### Returns

`any`

### Defined in

[packages/typedoc-plugin-markdown/src/theme/context/helpers/get-group-index.ts:8](https://github.com/typedoc2md/typedoc-plugin-markdown/blob/a350891d3362a78bb12907d480645f9c5cefd0d6/packages/typedoc-plugin-markdown/src/theme/context/helpers/get-group-index.ts#L8)

***

## getHierarchyType()

```ts
function getHierarchyType(
   this, 
   model, 
   options?): string
```

### Parameters

| Parameter           | Type                                                                  |
| ------------------- | --------------------------------------------------------------------- |
| `this`              | [`MarkdownThemeContext`](../../../../classes/MarkdownThemeContext.md) |
| `model`             | [`SomeType`](https://typedoc.org/api/classes/Models.SomeType.html)    |
| `options`?          | `object`                                                              |
| `options.isTarget`? | `boolean`                                                             |

### Returns

`string`

### Defined in

[packages/typedoc-plugin-markdown/src/theme/context/helpers/get-hierarchy-type.ts:5](https://github.com/typedoc2md/typedoc-plugin-markdown/blob/a350891d3362a78bb12907d480645f9c5cefd0d6/packages/typedoc-plugin-markdown/src/theme/context/helpers/get-hierarchy-type.ts#L5)

***

## getKeyword()

```ts
function getKeyword(model): string
```

### Parameters

| Parameter | Type                                                                           |
| --------- | ------------------------------------------------------------------------------ |
| `model`   | [`ReflectionKind`](https://typedoc.org/api/enums/Models.ReflectionKind-1.html) |

### Returns

`string`

### Defined in

[packages/typedoc-plugin-markdown/src/theme/context/helpers/get-keyword.ts:3](https://github.com/typedoc2md/typedoc-plugin-markdown/blob/a350891d3362a78bb12907d480645f9c5cefd0d6/packages/typedoc-plugin-markdown/src/theme/context/helpers/get-keyword.ts#L3)

***

## getModifier()

```ts
function getModifier(model): string | null
```

### Parameters

| Parameter | Type                                                                                         |
| --------- | -------------------------------------------------------------------------------------------- |
| `model`   | [`DeclarationReflection`](https://typedoc.org/api/classes/Models.DeclarationReflection.html) |

### Returns

`string` | `null`

### Defined in

[packages/typedoc-plugin-markdown/src/theme/context/helpers/get-modifier.ts:3](https://github.com/typedoc2md/typedoc-plugin-markdown/blob/a350891d3362a78bb12907d480645f9c5cefd0d6/packages/typedoc-plugin-markdown/src/theme/context/helpers/get-modifier.ts#L3)

***

## getParameterDefaultValue()

```ts
function getParameterDefaultValue(model): string
```

### Parameters

| Parameter | Type                                                                                     |
| --------- | ---------------------------------------------------------------------------------------- |
| `model`   | [`ParameterReflection`](https://typedoc.org/api/classes/Models.ParameterReflection.html) |

### Returns

`string`

### Defined in

[packages/typedoc-plugin-markdown/src/theme/context/helpers/get-parameter-default-value.ts:3](https://github.com/typedoc2md/typedoc-plugin-markdown/blob/a350891d3362a78bb12907d480645f9c5cefd0d6/packages/typedoc-plugin-markdown/src/theme/context/helpers/get-parameter-default-value.ts#L3)

***

## getProjectName()

```ts
function getProjectName(stringWithPlaceholders, page): string
```

### Parameters

| Parameter                | Type                                                                                                                                                              |
| ------------------------ | ----------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `stringWithPlaceholders` | `string`                                                                                                                                                          |
| `page`                   | [`MarkdownPageEvent`](../../../../../app/namespaces/events/classes/MarkdownPageEvent.md)\<[`Reflection`](https://typedoc.org/api/classes/Models.Reflection.html)> |

### Returns

`string`

### Defined in

[packages/typedoc-plugin-markdown/src/theme/context/helpers/get-project-name.ts:4](https://github.com/typedoc2md/typedoc-plugin-markdown/blob/a350891d3362a78bb12907d480645f9c5cefd0d6/packages/typedoc-plugin-markdown/src/theme/context/helpers/get-project-name.ts#L4)

***

## getPropertyDefaultValue()

```ts
function getPropertyDefaultValue(model): null | string
```

### Parameters

| Parameter | Type                                                                                         |
| --------- | -------------------------------------------------------------------------------------------- |
| `model`   | [`DeclarationReflection`](https://typedoc.org/api/classes/Models.DeclarationReflection.html) |

### Returns

`null` | `string`

### Defined in

[packages/typedoc-plugin-markdown/src/theme/context/helpers/get-property-default-value.ts:4](https://github.com/typedoc2md/typedoc-plugin-markdown/blob/a350891d3362a78bb12907d480645f9c5cefd0d6/packages/typedoc-plugin-markdown/src/theme/context/helpers/get-property-default-value.ts#L4)

***

## getReflectionFlags()

```ts
function getReflectionFlags(this, reflection): string[]
```

### Parameters

| Parameter    | Type                                                                                                                                                                                     |
| ------------ | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `this`       | [`MarkdownThemeContext`](../../../../classes/MarkdownThemeContext.md)                                                                                                                    |
| `reflection` | [`DeclarationReflection`](https://typedoc.org/api/classes/Models.DeclarationReflection.html) \| [`SignatureReflection`](https://typedoc.org/api/classes/Models.SignatureReflection.html) |

### Returns

`string`\[]

### Defined in

[packages/typedoc-plugin-markdown/src/theme/context/helpers/get-reflection-flags.ts:5](https://github.com/typedoc2md/typedoc-plugin-markdown/blob/a350891d3362a78bb12907d480645f9c5cefd0d6/packages/typedoc-plugin-markdown/src/theme/context/helpers/get-reflection-flags.ts#L5)

***

## getReturnType()

```ts
function getReturnType(this, model?): string
```

### Parameters

| Parameter | Type                                                                  |
| --------- | --------------------------------------------------------------------- |
| `this`    | [`MarkdownThemeContext`](../../../../classes/MarkdownThemeContext.md) |
| `model`?  | [`SomeType`](https://typedoc.org/api/classes/Models.SomeType.html)    |

### Returns

`string`

### Defined in

[packages/typedoc-plugin-markdown/src/theme/context/helpers/get-return-type.ts:5](https://github.com/typedoc2md/typedoc-plugin-markdown/blob/a350891d3362a78bb12907d480645f9c5cefd0d6/packages/typedoc-plugin-markdown/src/theme/context/helpers/get-return-type.ts#L5)

***

## isGroupKind()

```ts
function isGroupKind(model): boolean
```

### Parameters

| Parameter | Type                                                                                                                                                                                     |
| --------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `model`   | [`DeclarationReflection`](https://typedoc.org/api/classes/Models.DeclarationReflection.html) \| [`SignatureReflection`](https://typedoc.org/api/classes/Models.SignatureReflection.html) |

### Returns

`boolean`

### Defined in

[packages/typedoc-plugin-markdown/src/theme/context/helpers/is-group-kind.ts:7](https://github.com/typedoc2md/typedoc-plugin-markdown/blob/a350891d3362a78bb12907d480645f9c5cefd0d6/packages/typedoc-plugin-markdown/src/theme/context/helpers/is-group-kind.ts#L7)

***

## useTableFormat()

```ts
function useTableFormat(this, key): boolean
```

### Parameters

| Parameter | Type                                                                  |
| --------- | --------------------------------------------------------------------- |
| `this`    | [`MarkdownThemeContext`](../../../../classes/MarkdownThemeContext.md) |
| `key`     | `"parameters"` \| `"properties"` \| `"enums"` \| `"typeDeclarations"` |

### Returns

`boolean`

### Defined in

[packages/typedoc-plugin-markdown/src/theme/context/helpers/use-table.format.ts:3](https://github.com/typedoc2md/typedoc-plugin-markdown/blob/a350891d3362a78bb12907d480645f9c5cefd0d6/packages/typedoc-plugin-markdown/src/theme/context/helpers/use-table.format.ts#L3)
