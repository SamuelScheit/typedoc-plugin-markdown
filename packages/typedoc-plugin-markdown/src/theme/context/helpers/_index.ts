export * from './get-angle-bracket';
export * from './get-comment-parts';
export * from './get-declaration-type';
export * from './get-flattened-declarations';
export * from './get-hierarchy-type';
export * from './get-keyword';
export * from './get-modifier';
export * from './get-parameter-default-value';
export * from './get-project-name';
export * from './get-property-default-value';
export * from './get-reflection-flags';
export * from './get-return-type';
export * from './get-short-description';
export * from './index';
export * from './is-group-kind';
export * from './use-table.format';
