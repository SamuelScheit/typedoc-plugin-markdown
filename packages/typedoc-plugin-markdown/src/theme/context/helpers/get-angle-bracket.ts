import { MarkdownThemeContext } from '@plugin/theme';

export function getAngleBracket(
  this: MarkdownThemeContext,
  bracket: '<' | '>',
): string {
  const useEntities = this.options.getValue('useHTMLEncodedBrackets');
  if (bracket === '<') {
    return useEntities ? '&lt;' : '\\<';
  }
  return useEntities ? '&gt;' : '\\>';
}
