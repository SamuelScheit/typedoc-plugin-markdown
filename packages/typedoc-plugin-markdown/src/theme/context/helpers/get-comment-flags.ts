import { camelToTitleCase } from 'libs/utils';
import { MarkdownThemeContext } from 'theme';
import { DeclarationReflection, SignatureReflection } from 'typedoc';

export function getCommentFlags(
  this: MarkdownThemeContext,
  reflection: DeclarationReflection | SignatureReflection,
): string[] {
  const flagsNotRendered: `@${string}`[] = [
    '@showCategories',
    '@showGroups',
    '@hideCategories',
    '@hideGroups',
  ];
  const flags: string[] = [];
  if (reflection.comment) {
    for (const tag of reflection.comment.modifierTags) {
      if (!flagsNotRendered.includes(tag)) {
        flags.push(camelToTitleCase(tag.substring(1)));
      }
    }
  }
  return flags;
}
