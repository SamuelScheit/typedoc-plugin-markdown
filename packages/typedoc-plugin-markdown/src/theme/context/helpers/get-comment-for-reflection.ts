import { DeclarationReflection } from 'typedoc';

export function getCommentForReflection(model: DeclarationReflection) {
  return model.signatures?.length
    ? model.signatures[0].comment || model.comment
    : model.comment;
}
