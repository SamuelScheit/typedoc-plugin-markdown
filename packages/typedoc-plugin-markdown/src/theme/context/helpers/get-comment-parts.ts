import * as fs from 'fs';
import { link } from 'libs/markdown';
import { MarkdownThemeContext } from 'theme';
import { CommentDisplayPart, InlineTagDisplayPart } from 'typedoc';

export function getCommentParts(
  this: MarkdownThemeContext,
  model: CommentDisplayPart[],
): string {
  const md: string[] = [];
  for (const part of model) {
    switch (part.kind) {
      case 'text':
        md.push(part.text);
        break;
      case 'code':
        md.push(part.text);
        break;
      case 'inline-tag':
        switch (part.tag) {
          case '@label':
          case '@inheritdoc':
            break;
          case '@link':
          case '@linkcode':
          case '@linkplain': {
            if (part.target) {
              const url = getUrl(part);
              const wrap = part.tag === '@linkcode' ? '`' : '';
              md.push(
                url
                  ? `${link(`${wrap}${part.text}${wrap}`, this.getRelativeUrl(url))}`
                  : part.text,
              );
            } else {
              md.push(part.text);
            }
            break;
          }
          default:
            md.push(`{${part.tag} ${part.text}}`);
            break;
        }
        break;
      case 'relative-link':
        switch (typeof part.target) {
          case 'number':
            {
              const reflection = this.page.project.files.resolve(part.target);
              if (typeof reflection === 'object' && reflection.url) {
                md.push(this.getRelativeUrl(reflection.url));
                break;
              }
              const fileName = this.page.project.files.getName(part.target);
              if (fileName) {
                md.push(this.getRelativeUrl(`_media/${fileName}`));
                break;
              }
            }
            break;
          case 'undefined':
            md.push(part.text);
            break;
        }
        break;
      default:
        md.push('');
    }
  }
  return md.join('');
}

function getUrl(part: InlineTagDisplayPart) {
  if ((part.target as any).url) {
    return (part.target as any).url;
  }

  if ((part.target as any)?.parent?.url) {
    return (part.target as any)?.parent?.url;
  }

  return typeof part.target === 'string' ? part.target : '';
}

export function isFile(file: string) {
  try {
    return fs.statSync(file).isFile();
  } catch {
    return false;
  }
}
