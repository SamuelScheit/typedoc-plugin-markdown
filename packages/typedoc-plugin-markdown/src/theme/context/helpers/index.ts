/**
 * The `helpers` namespace holds the helpers for the theme and are smaller utility functions that return snippets or text or other data transformations.
 *
 * Please note that partials:
 *
 * - Take a `model` param (that references a specific TypeDoc model) and an `options` param if required.
 * - Can reference other helpers but should not reference partials.
 * - Can return strings or other models.
 *
 * @module
 */
export * from './_index';
