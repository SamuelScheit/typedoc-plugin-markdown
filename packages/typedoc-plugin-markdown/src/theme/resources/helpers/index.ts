export { flattenDeclarations } from './flatten-declarations';
export { getDeclarationType } from './get-declaration-type';
export { getIndexTitle } from './get-index-title';
export { getKeyword } from './get-keyword';
export { getModifier } from './get-modifier';
export { getParameterDefaultValue } from './get-parameter-default-value.';
export { isGroupKind } from './is-group-kind';
export { parseComments } from './parse-comments';
export { parseUrl } from './parse-url';