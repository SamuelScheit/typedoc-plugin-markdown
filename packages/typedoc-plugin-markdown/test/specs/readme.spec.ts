import {
  FixtureOutputDir,
  FixtureOutputFileStrategy,
} from '../__utils__/fixture-config';
import { expectFileToEqual } from '../__utils__/helpers';

describe(`Readmes`, () => {
  test(`should get merged readme for members`, () => {
    expectFileToEqual(
      FixtureOutputDir.Readme,
      FixtureOutputFileStrategy.Members,
      ['index.md'],
    );
  });

  test(`should get merged readme for modules`, () => {
    expectFileToEqual(
      FixtureOutputDir.Readme,
      FixtureOutputFileStrategy.Modules,
      ['index.md'],
    );
  });
});
