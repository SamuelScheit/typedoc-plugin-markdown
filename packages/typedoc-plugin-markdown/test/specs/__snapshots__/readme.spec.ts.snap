// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Readmes should get merged readme for members: (Output File Strategy "members") (Option Group "1") 1`] = `
"**typedoc-stubs** • API

***

# MY README

> Some block quote

Some text.

## Readme heading

Some text.

## typedoc-stubs

Module commments

### Functions

- [someFunction](functions/someFunction.md)
"
`;

exports[`Readmes should get merged readme for modules: (Output File Strategy "modules") (Option Group "1") 1`] = `
"**typedoc-stubs** • API

***

# MY README

> Some block quote

Some text.

## Readme heading

Some text.

## typedoc-stubs

Module commments

### Contents

- [Functions](index.md#functions)
    - [someFunction](index.md#somefunction)

### Functions

#### someFunction()

> **someFunction**(): \`void\`

##### Returns

\`void\`
"
`;
